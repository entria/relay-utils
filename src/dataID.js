/* @flow */
import { isValidObject } from './utils';

/**
 * Useful to remove __dataID__ to send data as mutation input
 */
export const removeDataID = (relayObj: Object) =>
  Object.keys(relayObj).filter(key => key !== '__dataID__').reduce((obj, key) => {
    const value = relayObj[key];
    const simpleValue = isValidObject(value) ? removeDataID(value) : value;

    return {
      ...obj,
      [key]: simpleValue,
    };
  }, {});
