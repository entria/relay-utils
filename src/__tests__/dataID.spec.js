/* @flow */
import { removeDataID } from '../dataID';

it('should remove __dataID__', () => {
  const data = {
    __dataID__: '123',
    name: 'name',
    description: 'description',
    deep: {
      __dataID__: '456',
      value: 10,
    },
  };

  const input = removeDataID(data);

  expect(input).toMatchSnapshot();
});
