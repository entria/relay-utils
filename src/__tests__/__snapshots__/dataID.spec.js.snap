// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should remove __dataID__ 1`] = `
Object {
  "deep": Object {
    "value": 10,
  },
  "description": "description",
  "name": "name",
}
`;
